tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/AsyncControllerActionInvoker_EndInvokeAction_Integration.cs

```diff
@@ -78,38 +81,64 @@ internal static CallTargetReturn<TResult> OnMethodEnd<TTarget, TResult>(TTarget
                     // us to defer finishing the span later while making sure callers of this method do not
                     // get this scope when calling Tracer.ActiveScope
                     var now = scope.Span.Context.TraceContext.Clock.UtcNow;
-                    httpContext.AddOnRequestCompleted(h => OnRequestCompletedAfterException(h, scope, now));
+                    httpContext.AddOnRequestCompleted(h => OnRequestCompletedAfterException(h, scope, proxyScope, now));
 
                     scope.SetFinishOnClose(false);
+                    proxyScope?.SetFinishOnClose(false); // TODO how do we sync this to finish at the same time as the above scope?
```

Comment:
Span has a `Finish` overload that takes a `DateTimeOffset`, so you can grab the value, and then set it on both scopes